src/Routes.js

```diff
@@ -1,17 +1,30 @@
-import React from "react";
+import React, { useContext } from "react";
 import { Route, Switch } from "react-router-dom";
 import Login from "./Components/Login/login";
+import Signup from "./Components/Login/signup";
 import Home from "./Components/Home/home";
+import Admin from "./Components/Admin/admin";
+import PasswordReset from "./Components/Login/passwordReset";
+import { UserContext } from "./Providers/UserProviders";
 
 export default function Routes() {
-    return (
+    const user = useContext(UserContext);
+    return user ? (
```

Comment:
When a user is logged in, and they go to any of the pages ("/" or "/login" or "/signup"), it displays the original not signed in content first, then after a couple of seconds, it switches to Admin dashboard. Is that something that can be fixed?

Reply:
Hmm I think this is happening because the user object isn't fully loaded yet, I'm taking a look a possible ways we could fix this. One way would be to render a loader until the object is loaded

Reply:
Yup a loader is something I have used before and maybe handy for other parts of the project as well. @ahmedhamodi what are your thoughts on this? Don't want to be a blocker, so adding a `loader` type functionality can be an entirely different ticket, to add that type of functionality to the project in general.

Reply:
Yeah a loader/spinner is the way to go. But can be addressed later

Reply:
Cool, in that case, PR looks good to me!